src/components/Filter.tsx

```diff
@@ -26,38 +27,55 @@ export default class Filter extends React.Component<FilterProps, FilterState> {
   constructor(props: FilterProps) {
     super(props);
     this.flip = this.flip.bind(this);
-    this.state = { flip: this.props.initialFlip };
+    this.state = { flip: this.props.initialFlip, panelOpen: false };
     this.applyFilter = this.applyFilter.bind(this);
+    this.handleFilterPanelClick = this.handleFilterPanelClick.bind(this);
   }
 
   /**
-  * applyFilter()
-  * @param {string} name The name of the attribute being turned on or off.  This doesn't
-  * affect whether we're filtering by the attribute's presence vs. by its absence; it just
-  * determines whether the filter should take this attribute into account in the first place.
-  */
+   * applyFilter()
+   * @param {string} name The name of the attribute being turned on or off.  This doesn't
+   * affect whether we're filtering by the attribute's presence vs. by its absence; it just
+   * determines whether the filter should take this attribute into account in the first place.
+   */
   applyFilter(name: string) {
     this.props.setFilter(name);
   }
 
   /**
-  * Toggle whether we're filtering via the specified attributes' presence vs. via their absence
-  */
+   * Toggle whether we're filtering via the specified attributes' presence vs. via their absence
+   */
   flip() {
     this.setState({ flip: !this.state.flip });
     this.props.flipFilter();
   }
 
+  handleFilterPanelClick() {
```

Comment:
Add a comment block with a short description of this function (similar to the other functions).

Reply:
Added!